File: src/Library/ObjectModel/MediaObject.vb
Public Class MediaObject
   Inherits Pixa.Soundbridge.Library.SoundbridgeListObject

   Private _server As MediaServer

   Friend Sub New(ByVal server As MediaServer, ByVal index As Integer, ByVal name As String)
      MyBase.New(server.Soundbridge, index, name)
      _server = server
   End Sub

   Public ReadOnly Property Server() As MediaServer
      Get
         Return _server
      End Get
   End Property
End Class

File: src/Pixa.Soundbridge/MediaObject.cs
namespace Pixa.Soundbridge
{
    public class MediaObject : SoundbridgeListObject
    {
        private MediaServer _server;

        internal MediaObject(MediaServer server, int index, string name) : base(server.Soundbridge, index, name)
        {
            _server = server;
        }

        public MediaServer Server
        {
            get
            {
                return _server;
            }
        }
    }
}